### PR TITLE
Fix vm discovery and power_control tests

### PR DESCRIPTION
**Pull request by dajohnso**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit fca6845be5dc8042991a2bcd75cbc1cc86d0f271:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/infrastructure/test_vm_discovery.py:7:1: E265 block comment should start with '# '
./cfme/tests/infrastructure/test_vm_power_control.py:5:1: F401 'VMwareProvider' imported but unused
./cfme/tests/infrastructure/test_vm_power_control.py:11:1: E265 block comment should start with '# '
./cfme/tests/infrastructure/test_vm_power_control.py:200:13: E265 block comment should start with '# '
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e525be453e0d82983eb7f925a714433786afbeec:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/infrastructure/test_vm_discovery.py:7:1: E265 block comment should start with '# '
./cfme/tests/infrastructure/test_vm_power_control.py:5:1: F401 'VMwareProvider' imported but unused
./cfme/tests/infrastructure/test_vm_power_control.py:11:1: E265 block comment should start with '# '
./cfme/tests/infrastructure/test_vm_power_control.py:200:13: E265 block comment should start with '# '
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ac9093e466b2b33d0c8922cc434a554d9a846027:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/infrastructure/test_vm_power_control.py:5:1: F401 'VMwareProvider' imported but unused
./cfme/tests/infrastructure/test_vm_power_control.py:11:1: E265 block comment should start with '# '
./cfme/tests/infrastructure/test_vm_power_control.py:200:13: E265 block comment should start with '# '
./cfme/tests/infrastructure/test_vm_discovery.py:7:1: E265 block comment should start with '# '
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 84c8c3f89f721c01e698d355ae96f28c8f797728:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/infrastructure/test_vm_power_control.py:11:1: E265 block comment should start with '# '
./cfme/tests/infrastructure/test_vm_discovery.py:7:1: E265 block comment should start with '# '
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6d97ecad9c7a9e3fb25f8eab16a97df0a898935f:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/tests/infrastructure/test_vm_discovery.py [hunk 1]
```diff
@@ -12,38 +14,54 @@ def pytest_generate_tests(metafunc):
...
     Tests whether cfme will discover a vm change
     (add/delete) without being manually refreshed.
     """
     vm = virtual_machines.Vm(vm_name, provider_crud)
-    request.addfinalizer(vm.delete_from_provider)
-    if not provider_mgmt.does_vm_exist(vm_name):
-        vm.create_on_provider()
+
+    def _cleanup():
+        vm.delete_from_provider
```
Comment (mfalesni) [inline cfme/tests/infrastructure/test_vm_discovery.py]: `()`

File: cfme/tests/infrastructure/test_vm_power_control.py [hunk 2]
```diff
@@ -64,8 +60,13 @@ def vm_name():
 def test_vm(request, provider_crud, provider_mgmt, vm_name):
     """Fixture to provision appliance to the provider being tested if necessary"""
     vm = Vm(vm_name, provider_crud)
+    logger.info("provider_key: {}".format(provider_crud.key))
 
-    request.addfinalizer(vm.delete_from_provider)
+    def _cleanup():
+        vm.delete_from_provider
```
Comment (mfalesni) [inline cfme/tests/infrastructure/test_vm_power_control.py]: `()` (pollutes providers)

File: fixtures/provider.py [hunk 3]
```diff
@@ -40,6 +40,15 @@ def setup_provider(provider_key):
     _setup_provider(provider_key)
 
 
+@pytest.fixture(scope='class')
+def setup_provider_clsscope(provider_key):
+    """Module-scoped fixture to set up a provider"""
+    try:
+        _setup_provider(provider_key)
+    except Exception:
+        pytest.set_trace()
```
Comment (mfalesni) [inline fixtures/provider.py]: Is it wise to have this code in "production"? It would stuck the test run.

**Comment by dajoRH** [comment]
Lint report for commit 6ef527e646a0adee9200be564aa8b23d411b88ba:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Pull request merged**